xonsh/completer.py

```diff
@@ -317,13 +333,15 @@ def module_complete(self, prefix):
 
     def _quote_paths(self, paths, start, end):
         out = set()
+        quotable = ' (){}-,+*'
```

Comment:
Also, I think that the `-` is a tad too aggressive.  Many files have dashes in them.  Probably the only characters that are needed in this are whitespace and whatever globbing and expandvars() use as syntax.  Namely, `" \t\r\n${}%*"`